lms/djangoapps/commerce/views.py

```diff
@@ -148,9 +149,37 @@ def checkout_cancel(_request):
 @login_required
 def checkout_receipt(request):
     """ Receipt view. """
+
+    page_title = _('Receipt')
+    is_payment_complete = True
+    is_cybersource = all(k in request.POST for k in ('signed_field_names', 'decision', 'reason_code'))
+    if is_cybersource and request.POST['decision'] != 'ACCEPT':
+        # Cybersource may redirect users to this view if it couldn't recover
+        # from an error while capturing payment info.
+        is_payment_complete = False
+        page_title = _('Payment Failed')
+        reason_code = request.POST['reason_code']
+        # if the problem was with the info submitted by the user, we present more detailed messages.
+        is_user_payment_error = (200 <= int(reason_code) <= 233) or int(reason_code) in (101, 102, 240)
```

Comment:
Future maintainers might appreciate a link to CyberSource's documentation in a comment here, so they can understand where these ranges came from.